src/fragments/lib/datastore/js/data-access/observe-update-snippet.mdx

```diff
@@ -0,0 +1,55 @@
+```ts
+// Example showing how to observe the model and keep state updated before
+// performing a save. This uses the useEffect React hook, but you can
+// substitute for a similar mechanism in your application lifecycle with
+// other frameworks.
+
+function App() {
+  const [todo, setTodo] = useState<Todo>();
+
+  useEffect(() => {
+    /**
+     * This keeps `todo` fresh.
+     */
+    const sub = DataStore.observeQuery(Todo, (c) =>
+      c.id("eq", "e4dd1dc5-e85c-4566-8aaa-54a801396456")
+    ).subscribe(({ items }) => {
+      setTodo(items[0]);
```

Comment:
I see the other Android/Flutter example using a Post, any reason this is diverging here?

Reply:
Done!